src/main/scala/edu/ie3/osmogrid/io/input/InputDataProvider.scala

```diff
@@ -6,31 +6,136 @@
 
 package edu.ie3.osmogrid.io.input
 
-import akka.actor.typed.Behavior
-import akka.actor.typed.ActorRef
-import akka.actor.typed.scaladsl.Behaviors
+import akka.actor.typed.scaladsl.{ActorContext, Behaviors, StashBuffer}
+import akka.actor.typed.{ActorRef, Behavior}
+import com.acervera.osm4scala.EntityIterator.fromPbf
+import com.acervera.osm4scala.model.{NodeEntity, RelationEntity, WayEntity}
 import edu.ie3.osmogrid.cfg.OsmoGridConfig
 import edu.ie3.osmogrid.guardian.OsmoGridGuardian.OsmoGridGuardianEvent
+import edu.ie3.osmogrid.model.{OsmoGridModel, PbfFilter}
+import edu.ie3.util.osm.model.OsmEntity.{Node, Relation, Way}
+import org.locationtech.jts.geom.{
+  Coordinate,
+  LinearRing,
+  Point,
+  Polygon,
+  PrecisionModel
+}
+
+import java.io.{FileInputStream, InputStream}
+import java.time.ZonedDateTime
+import java.util.UUID
+import scala.collection.mutable.ListBuffer
+import scala.util.{Failure, Success, Try, Using}
 
 object InputDataProvider {
 
+  // external requests
+  sealed trait InputRequest
+
+  final case class ReqOsm(
+      runId: UUID,
```

Comment:
AFAIK, the `runId` is not needed here, anymore, isn't it? If there is always one `InputDataProvider` per run, then there is no need to communicate it here, right?

Reply:
BTW: Also applies for the requests then. And - I don't know the purpose exactly - I think `InvalidOsmRequest` could then be obsolete as well, right?